Permanent Fix for Flaky Unit Test

The test_alias unit test for RediSearch has some behaviour that is randomly flaky, specifically with the alias updates.  PRs constantly randomly fail because of this. So far, we've attempted to solve these issues by adding sleeps that are specific to GitHub runners, but that isn't correct.

```
    @pytest.mark.redismod
    @skip_ifmodversion_lt("2.0.0", "search")
    def test_alias():
        index1 = getClient()
        index2 = getClient()
    
        index1.hset("index1:lonestar", mapping={"name": "lonestar"})
        index2.hset("index2:yogurt", mapping={"name": "yogurt"})
    
        if os.environ.get("GITHUB_WORKFLOW", None) is not None:
            time.sleep(2)
        else:
            time.sleep(5)
    
        def1 = IndexDefinition(prefix=["index1:"], score_field="name")
        def2 = IndexDefinition(prefix=["index2:"], score_field="name")
    
        ftindex1 = index1.ft("testAlias")
        ftindex2 = index1.ft("testAlias2")
        ftindex1.create_index((TextField("name"),), definition=def1)
        ftindex2.create_index((TextField("name"),), definition=def2)
    
        # CI is slower
        try:
            res = ftindex1.search("*").docs[0]
        except IndexError:
            time.sleep(5)
            res = ftindex1.search("*").docs[0]
        assert "index1:lonestar" == res.id
    
        # create alias and check for results
        ftindex1.aliasadd("spaceballs")
        alias_client = getClient().ft("spaceballs")
        res = alias_client.search("*").docs[0]
        assert "index1:lonestar" == res.id
    
        # Throw an exception when trying to add an alias that already exists
        with pytest.raises(Exception):
            ftindex2.aliasadd("spaceballs")
    
        # update alias and ensure new results
        ftindex2.aliasupdate("spaceballs")
        alias_client2 = getClient().ft("spaceballs")
    
        if os.environ.get("GITHUB_WORKFLOW", None) is not None:
            time.sleep(5)
    
>       res = alias_client2.search("*").docs[0]
E       IndexError: list index out of range
```

--- a/tests/test_search.py
+++ b/tests/test_search.py
@@ -646,11 +646,6 @@ def test_alias():
     index1.hset("index1:lonestar", mapping={"name": "lonestar"})
     index2.hset("index2:yogurt", mapping={"name": "yogurt"})
 
-    if os.environ.get("GITHUB_WORKFLOW", None) is not None:
-        time.sleep(2)
-    else:
-        time.sleep(5)
-
     def1 = IndexDefinition(prefix=["index1:"])
     def2 = IndexDefinition(prefix=["index2:"])
 
@@ -659,12 +654,6 @@ def test_alias():
     ftindex1.create_index((TextField("name"),), definition=def1)
     ftindex2.create_index((TextField("name"),), definition=def2)
 
-    # CI is slower
-    try:
-        res = ftindex1.search("*").docs[0]
-    except IndexError:
-        time.sleep(5)
-        res = ftindex1.search("*").docs[0]
     assert "index1:lonestar" == res.id
 
     # create alias and check for results
@@ -680,9 +669,6 @@ def test_alias():
     # update alias and ensure new results
     ftindex2.aliasupdate("spaceballs")
     alias_client2 = getClient().ft("spaceballs")
-
-    if os.environ.get("GITHUB_WORKFLOW", None) is not None:
-        time.sleep(5)
 
     res = alias_client2.search("*").docs[0]
     assert "index2:yogurt" == res.id

--- a/tests/test_search.py
+++ b/tests/test_search.py
@@ -654,6 +654,7 @@ def test_alias():
     ftindex1.create_index((TextField("name"),), definition=def1)
     ftindex2.create_index((TextField("name"),), definition=def2)
 
+    res = ftindex1.search("*").docs[0]
     assert "index1:lonestar" == res.id
 
     # create alias and check for results

--- a/tests/test_search.py
+++ b/tests/test_search.py
@@ -651,11 +651,11 @@ def test_alias():
     else:
         time.sleep(5)
 
-    def1 = IndexDefinition(prefix=["index1:"], score_field="name")
-    def2 = IndexDefinition(prefix=["index2:"], score_field="name")
+    def1 = IndexDefinition(prefix=["index1:"])
+    def2 = IndexDefinition(prefix=["index2:"])
 
     ftindex1 = index1.ft("testAlias")
-    ftindex2 = index1.ft("testAlias2")
+    ftindex2 = index2.ft("testAlias2")
     ftindex1.create_index((TextField("name"),), definition=def1)
     ftindex2.create_index((TextField("name"),), definition=def2)
 

